### PR TITLE
Update Page::processResponse() with better exceptions

### PR DESCRIPTION
**Pull request by ihumanable**

### CODE REVIEW

File: Twilio/Page.php [hunk 1]
```diff
@@ -40,7 +40,17 @@ public function __construct(Version $version, Response $response) {
...
+            $code = $response->getStatusCode();
+
+            $content = $response->getContent();
+
+            if (is_array($content)) {
+                $message .= isset($content['message']) ? ': ' . $content['message'] : '';
+                $code = isset($content['code']) ? $content['code'] : $code;
+            }
+
+            throw new DeserializeException($message, $code);
```
Comment (jingming) [inline Twilio/Page.php]: Should we throw a `RestException` here to be consistent with https://github.com/twilio/twilio-php/blob/master/Twilio/Version.php#L85?

**Comment by Gipetto** [comment]
+1

File: Twilio/Page.php (hunk 1 above)
Comment (ihumanable) [inline Twilio/Page.php]: We were throwing a DeserializeException before and that seems like an appropriate error to throw at this point in the lifecycle.

Comment (jingming) [inline Twilio/Page.php]: I feel like this one should throw a `RestException` as this is the bad status code response, where as `loadPage` should throw the `DeserializeException`

Comment (ihumanable) [inline Twilio/Page.php]: Ok, that makes sense, I'll make the change and call it out as a potentially backwards incompatibility (if you were explicitly catching `DeserializeException`)

**Comment by jingming** [comment]
:+1:

**Pull request merged**